lib/sax.js

```diff
@@ -719,6 +719,10 @@ function parseDoctypeInternalSubset(p, errorHandler) {
 				current = parsePI(p, errorHandler);
 			} else if (p.char() === '%') {
 				current = p.getMatch(g.PEReference);
+			} else if (p.char() === ']') {
```

Comment:
Once I'm back on my computer I will check if it can be refactored to avoid multiple occurrences of this condition, before landing it.